File: python/bob/bob.py
def response(hey_bob):
    """
    Determine Bob's response based on the given remark.

    Bob responds to remarks in the following ways:
    - If the remark is empty or contains only whitespace characters, Bob responds with "Fine. Be that way!"
    - If the remark is in all capital letters (YELLING), Bob responds with:
      - "Whoa, chill out!" if it's not a question
      - "Calm down, I know what I'm doing!" if it's a question
    - If the remark is a question, Bob responds with "Sure."
    - For any other remark, Bob responds with "Whatever."

    Parameters:
    remark (str): The remark or question given to Bob.

    Returns:
    str: Bob's response to the input remark.
    """
    # Check if the remark is empty or contains only whitespace characters
    if not hey_bob.strip():
        return "Fine. Be that way!"
    
    # Check if the remark is in all capital letters (YELLING)
    if hey_bob.isupper():
        # Check if the remark is a question (ends with a question mark)
        if hey_bob.endswith("?"):
            return "Calm down, I know what I'm doing!"
        else:
            return "Whoa, chill out!"
    
    # Check if the remark is a question
    if hey_bob.strip().endswith("?"):
        return "Sure."

    # Default response
    return "Whatever."

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  // Remove leading and trailing whitespace to check for empty message
  message = message.trim();

  // Check if the message is empty (no content)
  if (message === "") {
    return 'Fine. Be that way!';
  }

  // The message is a question (ends with a question mark)
  const isQuestion = message.endsWith('?');

  // The message is in all capitals (yelling)
  const isShouting = message === message.toUpperCase() && message !== message.toLowerCase();

  if (isQuestion && isShouting) {
    return "Calm down, I know what I'm doing!";
  }

  if (isQuestion) {
    return "Sure.";
  }

  // Check if the message is in all capitals (yelling)
  if (isShouting) {
    return 'Whoa, chill out!';
  }

  // Default response if none of the above conditions are met
  return 'Whatever.';
};

